Sources/ParseSwift/LiveQuery/ParseLiveQuery.swift

```diff
@@ -223,7 +223,7 @@ extension ParseLiveQuery {
 
     var reconnectInterval: Int {
         let min = NSDecimalNumber(decimal: Swift.min(30, pow(2, attempts) - 1))
-        return Int.random(in: 0 ..< Int(truncating: min))
+        return Int(truncating: min)
```

Comment:
What's the reason for changing this?

Reply:
As I mentioned [here](https://community.parseplatform.org/t/parseswift-sdk-observe-livequery-websocket-status/1739/35), I suspect a code race as the reconnectionInterval is mostly 0 for first two attempts and `resumeTask()` id deallocating the delegate with `URLSession.liveQuery.delegates.removeValue(forKey: self.task)` 

https://github.com/parse-community/Parse-Swift/blob/e3f51f4d5f1520d694edac248ebb883f2c05d375/Sources/ParseSwift/LiveQuery/ParseLiveQuery.swift#L229-L249

Is there a reason for the random Int? Because without the LiveQuery reconnection seems to behave much more reproducible and is even reconnection on server reset without holding the execution with a breakpoint.

Reply:
> As I mentioned here, I suspect a code race as the reconnectionInterval is mostly 0 for first two attempts and resumeTask() is deallocating the delegate with URLSession.liveQuery.delegates.removeValue(forKey: self.task)

This will need more justification and discussion before a change like this can be accepted (you can open an [issue](https://github.com/parse-community/Parse-Swift/issues) to begin that discussion. The first few attempts should come out to zero because it's a random increasing interval that increases based on the amount of attempts, so zero the first 2-3 attempts makes sense and represents trying to reconnect immediately after disconnection.

> Because without the random Int the LiveQuery reconnection seems to behave much more reproducible and is even reconnecting on server reset without holding the execution with a breakpoint.

This seems odd because your change makes the value `0`, but you mention the first few values from the `reconnectionInterval` are `0` which produces the same thing.

> Is there a reason for the random Int?

In networking you generally have some form of increasing backoff when attempting to reconnect.

I recommend changing this back to the original is even if there is a problem, this doesn't seem like the appropriate fix. The test case shows the randomInterval is doing what it's suppose to do. You can add a print statement to see the values if you want: 

https://github.com/parse-community/Parse-Swift/blob/e3f51f4d5f1520d694edac248ebb883f2c05d375/Tests/ParseSwiftTests/ParseLiveQueryTests.swift#L506-L518

Reply:
If zeros are the problem, you can make the following change:

```swift
let min = NSDecimalNumber(decimal: Swift.min(30, pow(2, attempts)))
return Int.random(in: 1 ..< Int(truncating: min))
```

Reply:
Sorry for misunderstanding, the 0 was my focus and I assumed that without the random Int it is still increasing backoff, so that was the reason for that change. But I tested your update [#209](https://github.com/parse-community/Parse-Swift/pull/209/commits/3d5729921c0376473ee4ea4a37ae04b2e506d074) and I can confirm that it solved even better the reconnection issue that should this commit resolve.